python/Ganga/Core/MonitoringComponent/Local_GangaMC_Service.py

```diff
@@ -394,39 +394,34 @@ def __init__(self, argDict, enabled=True, timeout=0):
         # record the time when this hook has been run
         self._lastRun = 0
 
-def resubmit_if_required(jobList_fromset):
 
-    # resubmit if required
+def resubmit_if_required(jobList_fromset):
     for j in jobList_fromset:
-
         if not j.do_auto_resubmit:
             continue
 
         if len(j.subjobs) == 0:
             try_resubmit = j.info.submit_counter <= config['MaxNumResubmits']
         else:
             # Check for max number of resubmissions
-            skip = False
-            for s in j.subjobs:
-                if s.info.submit_counter > config['MaxNumResubmits'] or s.status == "killed":
-                    skip = True
-
-                if skip:
-                    continue
-
-                num_com = len([s for s in j.subjobs if s.status in ['completed']])
-                num_fail = len([s for s in j.subjobs if s.status in ['failed']])
+            n_completed = len([sj for sj in j.subjobs if sj.status == 'completed'])
+            n_failed = len([sj for sj in j.subjobs if sj.status == 'failed'])
 
-                #log.critical('Checking failed subjobs for job %d... %d %s',j.id,num_com,num_fail)
-
-                try_resubmit = num_fail > 0 and (float(num_fail) / float(num_com + num_fail)) < config['MaxFracForResubmit']
+            # Check n_failed > 0 to avoid ZeroDivisionError
+            if n_failed > 0 and (float(n_failed) / float(n_completed + n_failed)) >= config['MaxFracForResubmit']:
+                # Too many subjobs have failed, stop trying
+                continue
 
-            if try_resubmit:
-                if j.backend.check_auto_resubmit():
-                    log.warning('Auto-resubmit job %d...' % j.id)
-                    j.auto_resubmit()
+            # Only submit if jobs have failed and not a single subjob has
+            # exceeded config['MaxNumResubmits']
+            try_resubmit = n_failed > 0 and not any([
+                sj.info.submit_counter > config['MaxNumResubmits']
```

Comment:
@chrisburr I'm just wondering slight;y if the `Job.py` is the best place to be accessing the config from the monitoring system. But if it works I'm happy 👍 
Good work for looking at this btw, as per #327 I didn't have a clue what the original code was doing and I think it needed a rewrite.

(As an aside)
I'm wondering what the _correct_ behaviour is for this, I simply don't know as I never used this.
Maybe we can introduce a 2nd max resubmit paramter into the configuration (https://github.com/ganga-devs/ganga/blob/develop/python/Ganga/__init__.py#L218), one parameter for the max resubmit per subjob and one for the resubmit for a master job?